### PR TITLE
Adding check for latest test agent version in deploy and run task

### PR DESCRIPTION
**Pull request by prmittal**

### CODE REVIEW

**Comment by AshwiniChalla** [comment]
test it on both 64bit and 32bit machines

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 1]
```diff
@@ -885,6 +885,40 @@ function InvokeTestAgentConfigExe([string[]] $Arguments, [string] $Version, [Sys
     throw "Did not find TestAgentConfig.exe at : $exePath. Ensure that TestAgent is installed."
 }
 
+function Locate-TestVersion()
```
Comment (allendm-msft) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: why repeat the code.. put it out into a common ps1

**Comment by prawalagarwal** [comment]
No description?

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 (hunk 1 above)
Comment (prmittal) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: On the build agent, we use content of script to execute on test agent and not the files themselves. Therefore, if functions are not all in same file, it won't work.

File: Tasks/RunDistributedTests/RunDistributedTests.ps1 [hunk 2]
```diff
@@ -85,13 +87,13 @@ if(![string]::IsNullOrWhiteSpace($testSuite))
...
+            Invoke-RunDistributedTests -TestMachineGroup $testMachineGroup -SourceFilter $sourcefilters -TestCaseFilter $testFilterCriteria -RunSettingsPath $runSettingsFilePreview -Platform $platform -Configuration $configuration -CodeCoverageEnabled $codeCoverageEnabledPreview -TestRunParams $overrideRunParamsPreview -TestDropLocation $dropLocation -Connection $connection -TestConfiguration $testConfigurations -AutMachineGroup $autMachineGroup -UnregisterTestAgentScriptLocation $unregisterTestAgentScriptLocation -TestRunTitle $testRunTitle -TestSelection $testSelection -TestPlan $testPlanId -TestSuites $testSuites -TestConfig $testConfigurationId -TaskContext $distributedTaskContext -CheckTestAgentCompatScriptLocation $checkTaCompatScriptLocation -HelperScriptLocation $helpers
         }
```
Comment (kaadhina) [inline Tasks/RunDistributedTests/RunDistributedTests.ps1]: We are passing the new parameter based on taskContextMemberExists. If Nitin's work goes in for 95, this condition will not reflect the presence/absence of the new parameter in 96.

File: Tasks/DeployVisualStudioTestAgent/DeployTestAgent.ps1 [hunk 3]
```diff
@@ -90,7 +93,7 @@ if($taskContextMemberExists){
...
+    Invoke-DeployTestAgent -TaskContext $distributedTaskContext -MachineEnvironment $environment -UserName $machineUserName -Password $machinePassword -MachineNames $testMachineGroup -RunAsProcess $runAsProcess -LogonAutomatically $logonAutomatically -DisableScreenSaver $disableScreenSaver -AgentLocation $agentLocation -UpdateTestAgent $updateTestAgent -InstallAgentScriptLocation $installAgentScriptLocation -ConfigureTestAgentScriptLocation $configureTestAgentScriptLocation -CheckAgentInstallationScriptLocation $checkAgentInstallationScriptLocation -downloadTestAgentScriptLocation $downloadTestAgentScriptLocation -Connection $connection -PersonalAccessToken $personalAccessToken -DataCollectionOnly $isDataCollectionOnly -VerifyTestMachinesAreInUseScriptLocation $verifyTestMachinesAreInUse -HelperScriptLocation $helpers
```
Comment (kaadhina) [inline Tasks/DeployVisualStudioTestAgent/DeployTestAgent.ps1]: compat with old build agent?

**Pull request closed without merging**